### PR TITLE
cleanup: f-string formatting

### LINKED ISSUE #785: Let's use f-strings instead of .format() to increase readability of documentation
Functions in this repo appear in the official documentation. All of them I've seen so far uses the `.format()` method of string processing, instead of the much more readable f-strings. I am able to change all `.format()`s to f-strings if this is accepted as a better solution.

### PR DESCRIPTION
**Pull request by zabop**
As discussed [here](https://github.com/googleapis/python-storage/issues/785), code in this repo appears directly in the documentation. (An [example](https://cloud.google.com/storage/docs/creating-buckets#prereq-code-samples), there are hundreds.) This commit introduces f-strings to increase readability of both code and documentation.

Fixes #785 🦕

### CODE REVIEW

File: docs/snippets.py
Comment (Mariatta) [inline docs/snippets.py:302]: The conversion here doesn't look quite right.
`%-25s` adds 25 space padding to the first item from the tuple. 
Suggested alternative:
```
name, doc: _name_and_doc(example)
print(f"{name:>25}: {doc}"
```

**Comment by zabop** [comment]
I am not sure why 2 tests failed, let me know if further input is needed from me.

File: samples/snippets/notification_polling.py
Comment (Mariatta) [inline samples/snippets/notification_polling.py:79]: Looks like you missed converting the one below (line 83) :)

File: tests/unit/test_blob.py
Comment (Mariatta) [inline tests/unit/test_blob.py:2876]: I think the `:d` format specifier is not needed anymore

```suggestion
            content_range = f"bytes 0-{blob.chunk_size - 1:}/*"
```

File: tests/unit/test_blob.py
Comment (Mariatta) [inline tests/unit/test_blob.py:2878]:
```suggestion
            content_range = f"bytes 0-{blob.chunk_size - 1}/{size}"
```

**Comment by Breathtender** [comment]
Yep, for the conventionalcommits message, you just need to amend the commit message to be "cleanup: f-string formatting". I thought editing the title there would do it, but seems not.
For the other one, you can run `nox -s blacken lint` and it'll lint cleanup all the files.
If you're not able to do either of those, let me know and I can help.

**Review by Mariatta — CHANGES_REQUESTED** [review]
Thank you. Mostly looks correct, except for a few spots.

**Comment by zabop** [comment]
@Mariatta, thank you, great catches!
@Breathtender, I get an error with `nox`:

```
% nox -s blacken lint
nox > Running session blacken
nox > Session blacken skipped: Python interpreter 3.8 not found.
nox > Running session lint
nox > Session lint skipped: Python interpreter 3.8 not found.
nox > Ran multiple sessions:
nox > * blacken: skipped
nox > * lint: skipped

```

Which I do not yet know how to solve. my Python interpreter seems to be Python 3.8 (screenshot of pottom right corner of PyCharm):

image

**Comment by Breathtender** [comment]
Oh that's strange. I'll check on why that's happening. For now I'll just run the lint myself and add the commit, once you push up your changes to address @Mariatta's comments.

**Comment by zabop** [comment]
@Breathtender, just read your reply now, sorry; in the meantime I figured out what's going on and were able to run `nox -s blacken lint`. Just pushed the results.

**Comment by zabop** [comment]
Addressing @Mariatta suggestions now...

**Comment by zabop** [comment]
I think I have addressed the suggestions.

My last commit was the result of a typo & forgetting to run `nox -s blacken lint` before. Do I have to keep the `cleanup: f-string formatting` commit messages when I push fixes to an already open PR, or is more freedom allowed here? (To rephrase the question: which commit message is checked by the conventionalcommits test?) I used the `remove unnecessary :d specifier` commit message to approve @Mariatta suggestions, I hope that's ok.

**Comment by Breathtender** [comment]
No, it'll automatically roll all your commits together, and so the first commit message is ultimately the one that matters.

**Comment by Breathtender** [comment]
Thanks for making those adjustments! I'll wait for @Mariatta's approval on the changes requested, but this is looking good to me.

**Review by Mariatta — APPROVED** [review]
Thanks! LGTM

**Review by unforced — APPROVED** [review]

**Pull request merged**
